frappe/core/doctype/user/user.js

```diff
@@ -192,6 +192,15 @@ frappe.ui.form.on('User', {
 	}
 })
 
+function has_access_to_edit_user() {
+	return has_common(frappe.user_roles, get_roles_for_editing_user());
+}
+
+function get_roles_for_editing_user() {
+	return frappe.get_meta('User').permissions
+		.filter(perm => perm.permlevel >= 1 && perm.write)
```

Comment:
I would still go for the hard-coded setup via System Manager as this is

1. Hard to discover
2. Still possible to get locked out of the system.